lib/rom/reader.rb

```diff
@@ -58,6 +58,36 @@ def each
       mapper.process(relation) { |tuple| yield(tuple) }
     end
 
+    # Returns a single tuple from the relation if there is one.
+    #
+    # @raise [ROM::TupleCountMismatchError] if the relation contains more than
+    #   one tuple
+    #
+    # @api public
+    def one
+      if relation.count > 1
+        raise(
+          TupleCountMismatchError,
+          'The relation consists of more than one tuple'
+        )
+      else
+        mapper.process(relation).first
+      end
+    end
```

Comment:
That's an interesting situation - this introduces a requirement that every relation must respond to `count` (provided by the adapter). Commands also require this interface but for simple read-only data sources an adapter wouldn't have to care about that. If we use `count` here we'd introduce a new required interface for *all* adapters. Hmmmm